### PR TITLE
Files galore

### PR DESCRIPTION
**Pull request by QuantumlyTangled**
idek what to say really.

### CODE REVIEW

**Comment by Soumil07** [comment]
Can you lint src/index off the linter rules you added?

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+```
+
+<!--
+If you issue is a proposal, you can provide its use case showing code.
+Make sure to explain everything well and comment the code, so we can
+follow up.
+-->
+
+### Expected an actual behaviour
```
Comment (Soumil07) [inline .github/ISSUE_TEMPLATE.md]: Expected actual behavior

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Contributing
+
+To contribute to this repository, feel free to create a new fork of the repository
+submit a pull request. We highly suggest [ESLint](https://eslint.org/) to be installed
```
Comment (Soumil07) [inline .github/CONTRIBUTING.md]: **and** submit a new pull request...

File: src/index.js [hunk 3]
```diff
@@ -1,41 +1,40 @@
...
+const { get } = require('snekfetch'); // eslint-disable-line
 
 module.exports = class Clash {
 
-  constructor(key) {
-    key = key || process.env.COC_API;
-    if (!key) throw new TypeError('You must provide a key while initializing this class or set a key in your environment variables.');
-                                                                                                 
-    /**
+    constructor(key) {
```
Comment (MrJacz) [inline src/index.js]: Place the process.env in the constructor aka `key = process.env.shit`

File: src/index.js [hunk 4]
```diff
@@ -1,41 +1,40 @@
...
+
+        /**
     * The base URL for all API endpoints
     * @since 0.0.1
     */
-    this.base_URL = 'https://api.clashofclans.com/v1/';
-  }
-  
-  /** 
+        this.base_URL = 'https://api.clashofclans.com/v1/'; // eslint-disable-line
```
Comment (MrJacz) [inline src/index.js]: Stop using eslint-disable-line and dont use camel case

File: src/index.js [hunk 5]
```diff
@@ -1,41 +1,40 @@
...
-    const _tag = encodeURIComponent(tag).toUpperCase();
-    const { body: player } = await get(`${this.base_URL}players/${_tag.toString()}`)
-      .set('authorization', `Bearer ${this.key}`)
-      .catch(e => { throw e });
-    
-    if (player) return player;
-  }
-  
-  /**
+    async getPlayer(tag) { // eslint-disable-line
```
Comment (MrJacz) [inline src/index.js]: Why are you putting `// eslint-disable-line` here

File: src/index.js [hunk 6]
```diff
@@ -54,20 +53,21 @@ module.exports = class Clash {
...
-     const _tag = encodeURIComponent(tag).toUpperCase();
-     const { body: clan } = await get(`${this.base_URL}clans/${_tag.toString()}`)
-       .set('authorization', `Bearer ${this.key}`)
-       .catch(e => { throw e });
-    
-    if (clan) return clan;
-  }
-};
\ No newline at end of file
+    async getClan(tag) { // eslint-disable-line
```
Comment (MrJacz) [inline src/index.js]: Same here

**Comment by MrJacz** [comment]
I recommend you make a private get method to handle api interactions likehttps://github.com/MrJacz/tatsumaki.js/blob/master/src/lib/client.js#L56-L68

**Pull request merged**